src/modules/expression/interpolated_region.rs

```diff
@@ -38,36 +38,43 @@ fn is_escaped(word: &str, symbol: char) -> bool {
 }
 
 /// Parse Amber code's escaped strings and reterns it.
-fn parse_escaped_string(string: String, region_type: &InterpolatedRegionType) -> String {
-    let mut chars = string.chars().peekable();
+fn parse_escaped_string(meta: &mut ParserMetadata, string: String, region_type: &InterpolatedRegionType) -> String {
+    let mut chars = string.chars().enumerate().peekable();
     let mut result = String::new();
-    while let Some(c) = chars.next() {
+    while let Some((_, c)) = chars.next() {
         if c == '\\' {
             match chars.peek() {
-                Some('\n') => {}
-                Some('\\') => result.push('\\'),
-                Some('n') => result.push('\n'),
-                Some('t') => result.push('\t'),
-                Some('r') => result.push('\r'),
-                Some('0') => result.push('\0'),
-                Some('{') => result.push('{'),
-                Some('"') => {
-                    if *region_type == InterpolatedRegionType::Text {
-                        result.push('"');
-                    } else {
-                        result.push(c);
-                        continue;
-                    }
-                }
-                Some('$') => {
-                    if *region_type == InterpolatedRegionType::Command {
-                        result.push('$');
-                    } else {
-                        result.push(c);
-                        continue;
+                Some((_, '\n')) => {}
+                Some((_, '\\')) => result.push('\\'),
+                Some((_, 'n')) => result.push('\n'),
+                Some((_, 't')) => result.push('\t'),
+                Some((_, 'r')) => result.push('\r'),
+                Some((_, '0')) => result.push('\0'),
+                Some((_, '{')) => result.push('{'),
+                Some((_, '"')) if *region_type == InterpolatedRegionType::Text => result.push('"'),
+                Some((_, '$')) if *region_type == InterpolatedRegionType::Command => result.push('$'),
+                Some((i, peek)) => {
+                    let warning_msg = format!("Invalid escape sequence '\\{peek}'");
+                    let pos = PositionInfo::from_token(meta, meta.get_token_at(meta.get_index().saturating_sub(1)));
+                    if let Position::Pos(row, col) = pos.position {
+                        let c_pos = PositionInfo::at_pos(pos.path, (row, col + i ), 2);
```

Comment:
- [ ] test this:

- The inception
- ```
  echo "one
  two\a"
  ```
- ```
  trust $ if true; then
  echo \"true\"! fi $
  ```

Reply:
Tested

```
echo "lorem ipsum doloret amat\x10"
trust $ echo \"Hello\" $
echo "one
two\a"
trust $ if true; then
echo \"true\"; fi $
trust $ echo { trust $ echo { trust $ echo { trust $ echo Hello \"World\" $ } $ } $ } $
```

<img width="757" height="942" alt="image" src="https://github.com/user-attachments/assets/2b42ec00-ce7f-41ca-bbcf-d076077823a8" />